packages/replay-next/src/suspense/ObjectPreviews.ts

```diff
@@ -1,288 +1,87 @@
-import { Object, ObjectId, PauseId, Value as ProtocolValue } from "@replayio/protocol";
 import {
-  Deferred,
-  Record,
-  ResolvedRecord,
-  createDeferred,
-  createPendingRecord,
-  createResolvedRecord,
-  isPendingRecord,
-  isRejectedRecord,
-  isResolvedRecord,
-  updateRecordToRejected,
-  updateRecordToResolved,
-} from "suspense";
+  Object,
+  ObjectId,
+  ObjectPreviewLevel,
+  PauseId,
+  Value as ProtocolValue,
+} from "@replayio/protocol";
+import { Cache, createCache } from "suspense";
 
 import { ReplayClientInterface } from "../../../shared/client/types";
-import { createFetchAsyncFromFetchSuspense } from "../utils/suspense";
 import { cachePauseData } from "./PauseCache";
 
-type ObjectMap = Map<ObjectId, Object>;
-type RecordMap = Map<ObjectId, Record<Object>>;
-type PropertyRecordMap = Map<string, Record<ProtocolValue>>;
+export const objectCache: Cache<
+  [
+    client: ReplayClientInterface,
+    pauseId: PauseId,
+    objectId: ObjectId,
+    previewLevel: ObjectPreviewLevel
+  ],
+  Object
+> = createCache({
+  // The protocol only sends objects once per session;
+  // disable weak ref caching to ensure we don't lose data
+  config: { useWeakRef: false },
+  debugLabel: "objectCache",
+  getKey: ([client, pauseId, objectId, previewLevel]) => `${pauseId}:${objectId}:${previewLevel}`,
```

Comment:
:speech_balloon: Looking at https://suspense.vercel.app/createCache , I don't see any actual _reference_ docs or TS types describing the possible args to `createCache`, or what the resulting `Cache` object contains. It would be helpful to add those when you have time.

(related: I'm confused why `getKey()`'s first arg is an array of the arguments, instead of the separate args themselves directly.)

Reply:
Ah, it's not an "args array" - it's using named parameters. So `getKey` comes earlier because I alpha-sort keys by habit. 😄 

Good point on explicitly calling out the generic types and how they relate to the `getKey` and `load` params though.